django/db/migrations/autodetector.py

```diff
@@ -576,6 +583,107 @@ def generate_renamed_models(self):
                             self.old_model_keys.add((app_label, model_name))
                             break
 
+    def generate_moved_models(self):
+        """
+        Find any moved models, generate the operations for them, and remove
+        the old entry from the model lists. Must be run before other
+        model-level generation.
+        """
+        old_keys = self.old_model_keys | self.old_unmanaged_keys
+        added_models = self.new_model_keys - old_keys
+        added_unmanaged_models = self.new_unmanaged_keys - old_keys
```

Comment:
Is there a reason we even consider unmanaged models?

It feels like only considering moved models that are managed (from and to app) would simplify the implementation considerably and moving unmanaged models is really an edge case that we could add support for latter on.